src/SignalR/clients/ts/signalr/src/FetchHttpClient.ts

```diff
@@ -0,0 +1,123 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+import { AbortError, HttpError, TimeoutError } from "./Errors";
+import { HttpClient, HttpRequest, HttpResponse } from "./HttpClient";
+import { ILogger, LogLevel } from "./ILogger";
+
+export class FetchHttpClient extends HttpClient {
+  private readonly logger: ILogger;
+
+  public constructor(logger: ILogger) {
+    super();
+    this.logger = logger;
+  }
+
+  /** @inheritDoc */
+  public send(request: HttpRequest): Promise<HttpResponse> {
+    // Check that abort was not signaled before calling send
+    if (request.abortSignal && request.abortSignal.aborted) {
+      return Promise.reject(new AbortError());
+    }
+
+    if (!request.method) {
+      return Promise.reject(new Error("No method defined."));
+    }
+    if (!request.url) {
+      return Promise.reject(new Error("No url defined."));
+    }
+
+    return new Promise<HttpResponse>((resolve, reject) => {
+      // https://developers.google.com/web/updates/2017/09/abortable-fetch
+      const controller = new AbortController();
+      const signal = controller.signal;
+
+      const requestInfo: RequestInit = {
+        body: request.content || "",
+        credentials: "include",
+        headers: {
+          "Content-Type": "text/plain;charset=UTF-8",
```

Comment:
@Tratcher we need to update the auth code to handle this case as well. Fetch doesn’t set the headers we expect and of course we can set it but it’s clear that it’s not the right way to check

Reply:
What else can auth check?

Reply:
A header that doesn’t assume XMLHttRequest something with a better semantic name that means don’t redirect. In essence, it’s a feature of the cookie auth instead of Ajax detection. X-AspNetCore-DisableCookieAuthRedirect